### ISSUE TITLE
log: flag "log-dir-max-size" not work to SecondaryLogger

### ISSUE THREAD

**Issue by seekingua** [cockroachdb/cockroach#40827]
**Describe the problem**

when start cockroach using flag "log-dir-max-size", it works for mail logger but not for secondary logger

**To Reproduce**

1. start cockroach using:
```
start 
--insecure 
--store=/cockroach-data/store1 
--sql-audit-dir=/cockroach-data/store1_audit 
--log-file-max-size=100k 
--log-dir-max-size=300k
```

2. login by cockroach sql, create test table and turn on audit on table:
```
create table log(id int);
alter table log experimental_audit set read write;
```

3. generate more audit data
```
if db, err := sql.Open("postgres", "postgres://root@localhost:26257/defaultdb?sslmode=disable"); err == nil {
	query := "select * from log"
	for i := 1; i < 10000; i++ {
		_, err := db.Exec(query)
		if err != nil {
			log.Printf("%s", err)
		}
	}
	if err := db.Close(); err != nil {
		log.Printf("%s", err)
	}
}
```
4. list log file in directory: "/cockroach-data/store1_audit", all log files are retained and won't be deleted even if dir max size is larger than value of  flag '--log-dir-max-size'

```
-rw-r--r--  1 seekingua  admin  99945 Sep 17 22:56 ___server_start-sql-audit.seekingua.seekingua.2019-09-17T14_55_56Z.013403.log
-rw-r--r--  1 seekingua  admin  99945 Sep 17 22:56 ___server_start-sql-audit.seekingua.seekingua.2019-09-17T14_56_03Z.013403.log
-rw-r--r--  1 seekingua  admin  99945 Sep 17 22:56 ___server_start-sql-audit.seekingua.seekingua.2019-09-17T14_56_11Z.013403.log
-rw-r--r--  1 seekingua  admin  99945 Sep 17 22:56 ___server_start-sql-audit.seekingua.seekingua.2019-09-17T14_56_18Z.013403.log
-rw-r--r--  1 seekingua  admin  99945 Sep 17 22:56 ___server_start-sql-audit.seekingua.seekingua.2019-09-17T14_56_26Z.013403.log
-rw-r--r--  1 seekingua  admin  99945 Sep 17 22:56 ___server_start-sql-audit.seekingua.seekingua.2019-09-17T14_56_33Z.013403.log
-rw-r--r--  1 seekingua  admin  99945 Sep 17 22:56 ___server_start-sql-audit.seekingua.seekingua.2019-09-17T14_56_41Z.013403.log
-rw-r--r--  1 seekingua  admin  20073 Sep 17 22:56 ___server_start-sql-audit.seekingua.seekingua.2019-09-17T14_56_48Z.013403.log
```

**Expected behavior**
flag "--log-file-max-size" and "--log-dir-max-size" work for both main and secondary logger

**Environment:**
 - CockroachDB: master or v19.1.4
 - Server OS: macos 10.14.6
 - Client app:  cockroach sql

**Comment by seekingua** [comment]
#40835 fix this

**Comment by knz** [comment]
This particular issue was fixed in #41020.
